pl_bolts/datamodules/emnist_datamodule.py

```diff
@@ -0,0 +1,240 @@
+        if not any(s == self.split for s in self.dataset_cls.splits):
+            allowed_values = ', '.join([f'"{v}"' for v in self.dataset_cls.splits])
+            raise ValueError(
+                f'Invalid value provided for split (="{self.split}"). ' + f'Allowed splits are: {allowed_values}'
+            )
+        return True
+
+    @property
+    def split_metadata(self):
+        return self.dataset_cls._metadata.get('splits').get(self.split)
+
+    @property
+    def num_classes(self) -> int:
+        """
+        Return:
+           - for ``byclass``: 62
+           - for ``bymerge``: 47
+           - for ``balanced``: 47
+           - for ``digits``: 10
+           - for ``letters``: 47
+           - for ``mnist``: 10
+        """
+        # The _metadata is only added to EMNIST dataset
+        # to get split-specific metadata.
+        return self.split_metadata.get('num_classes')
+
+    def prepare_data(self, *args: Any, **kwargs: Any) -> None:
+        """
+        Saves files to data_dir
+        """
+
+        def _prepare_with_splits(split: str):
+            self.dataset_cls(self.data_dir, split=split, train=True, download=True)
+            self.dataset_cls(self.data_dir, split=split, train=False, download=True)
+
+        _prepare_with_splits(self.split)
+
+    def setup(self, stage: Optional[str] = None) -> None:
+        """
+        Creates train, val, and test dataset
+        """
+
+        # TODO: change type: Any to something like torch
+        def _setup_with_splits(split: str, train: bool, transform: Any):  # type: ignore[misc]
+            return self.dataset_cls(self.data_dir, split=split, train=train, transform=transform, **self.EXTRA_ARGS)
+
+        if stage == "fit" or stage is None:
+            train_transforms = self.default_transforms() if self.train_transforms is None else self.train_transforms
+            val_transforms = self.default_transforms() if self.val_transforms is None else self.val_transforms
+
+            dataset_train = _setup_with_splits(split=self.split, train=True, transform=train_transforms)
+
+            dataset_val = _setup_with_splits(split=self.split, train=True, transform=val_transforms)
+
+            # Split
+            self.dataset_train = self._split_dataset(dataset_train)
+            self.dataset_val = self._split_dataset(dataset_val, train=False)
+
+        if stage == "test" or stage is None:
+            test_transforms = self.default_transforms() if self.test_transforms is None else self.test_transforms
+            self.dataset_test = _setup_with_splits(split=self.split, train=False, transform=test_transforms)
+
+    def default_transforms(self) -> Callable:
+        if self.normalize:
+            emnist_transforms = transform_lib.Compose([
+                transform_lib.ToTensor(),
+                transform_lib.Normalize(mean=(0.5, ), std=(0.5, )),
+                # TODO: check that EMNIST also uses mean=0.5 and std=0.5
```

Comment:
Could you resolve this TODO?

Reply:
@akihironitta I didn't get much time today to work on this one. Let me take a look at it tomorrow/the day after.

Reply:
@akihironitta Please have a look at [this](https://github.com/PyTorchLightning/lightning-bolts/pull/676#issuecomment-877944327). :star:

Reply:
@sugatoray It looks great! I just directly updated your branch with a commit adding `emnist_normalization` using the stats (on train sets) you've shared. Mind having a look at the change (955d5b6)? If you find it reasonable, let's use it for `EMNISTDataModule.default_transforms()`.

Reply:
@akihironitta Thank you for making the update. I haven't checked the docs yet. Did you add these stats in the Binary/EMNISTDataModule docs as well? I think the user should be made aware of these values in the docstring and the example for the default_transforms is where probably making this available could be more helpful for the user. If you have not added this to the docstring yet, would you mind doing that as well?

**A note on choosing `mean = std = 0.5`**: Since the pixel values range between `[0.0, 1.0]`, a transformation such as: `(px - mean) / std` would only scale and shift the pixel values to a range of `[-1.0, 1.0]`. Typically, if you use a model which was already trained with some data and had you used a transformation with some mean and std, later on while using the same model, you would want to use the same mean/std for mimicking the previous data-transformations. However, in this case, the user is going to create a Datamodule and most likely train a model. So, it would not matter if a mean/std of 0.5/0.5 or some other value(s) is/are chosen (although, this may alter the rate of convergence). So, I think that we could be okay with even using 0.5 as the default values for mean and std. But, since we have it calculated already, we might as well provide the mean and std for all the splits in the documentation, in case the user wants to put them to use instead of 0.5. 

**One more thing**: I think this being a statistic of the `EMNIST` dataset, this Mean and SD information should be made part of the `EMNIST_METADATA` and then accessed from there. Whether you use them in the datamodules or not, these are effectively evaluated "facts" about the dataset and so should be part of the `EMNIST_METADATA` and thus will be available in `EMNIST._metadata`.

:bulb: What do you think?

Reply:
@akihironitta I added the `emnist_normalization()` into emnist_datamodules.py.